framework/src/problems/EigenProblem.C

```diff
@@ -513,8 +507,12 @@ EigenProblem::solve()
     // could rebuild matrices due to mesh changes or something else.
     _nl_eigen->attachSLEPcCallbacks();
 
-    // Scale eigen vector if necessary
-    preScaleEigenVector();
+    // If there is an eigenvalue, we scale 1/|Bx| to eigenvalue
+    if (_active_eigen_index < _nl_eigen->getNumConvergedEigenvalues())
+    {
+      std::pair<Real, Real> eig = _nl_eigen->getConvergedEigenvalue(_active_eigen_index);
+      preScaleEigenVector(eig);
+    }
```

Comment:
I am not sure if this is a good idea to overwrite what users pass in. For example, if users ask for "initial_eigenvalue = 1000", but we ignore their requirement  use the eigenvalue from the previous solve. Users will say that the code is wrong just like you said in the slack. 

If we intend to allow users to modify the initial eigenvalue, we should call a particular function on `EXEC_INITIAL`, while we do not want them touch that during Picard iteration and between Power and Newton

Reply:
We will not because the first time we are getting here, we do not have any valid eigenvalues on the solver side.

Reply:
The second time getting here, we should not use the initial_guess anyway.

Reply:
> he second time getting here, we should not use the initial_guess anyway.

It will depend on how do you get there.  I do feel wrong by ignoring the pass-in parameter. We might be explicit on that 

I think we could refactor this function a bit.  One function should be like this:

```
  if (exec_type == EXEC_INITIAL)
    preScaleEigenVector(_initial_eigenvalue);
```

Another one would be:

`preScaleEigenVector()` without an input parameter, where we will use the computed eigenvalue to scale the eigenvector

Reply:
In your `preScaleEigenVector()`, you will need to check the availability of the computed eigenvalue. We are doing the check and getting the eigenvalue outside here. I do not see the difference.

`initial_guess` is ONLY for the initialization in `EigenProblem::execute(EXEC_INITIAL)`.

Reply:
Github messed up. I saw the wrong code -:)

Reply:
I see the correct code now ....